nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/util/PCAP.java

```diff
@@ -0,0 +1,717 @@
+        }
+
+        /**
+         * Correction time in seconds between UTC and the local
+         * timezone of the following packet header timestamps.
+         */
+        public int thiszone() {
+            return thiszone;
+        }
+
+        /**
+         * In theory, the accuracy of time stamps in the capture; in
+         * practice, all tools set it to 0.
+         */
+        public long sigfigs() {
+            return sigfigs;
+        }
+
+        /**
+         * The "snapshot length" for the capture (typically 65535 or
+         * even more, but might be limited by the user), see: incl_len
+         * vs. orig_len.
+         */
+        public long snaplen() {
+            return snaplen;
+        }
+
+        /**
+         * Link-layer header type, specifying the type of headers at
+         * the beginning of the packet.
+         */
+        public Linktype network() {
+            return network;
+        }
+
+        public PCAP root() {
+            return root;
+        }
+
+        public PCAP parent() {
+            return parent;
+        }
+    }
+
+    /**
+     * @see <a href=
+     *      "https://wiki.wireshark.org/Development/LibpcapFileFormat#Record_.28Packet.29_Header">Source</a>
+     */
+    public static class Packet {
+        public ByteBufferInterface io;
+
+        public Packet(ByteBufferInterface io) {
+            this(io, null, null);
+        }
+
+        public Packet(ByteBufferInterface io, PCAP parent) {
+            this(io, parent, null);
+        }
+
+        public Packet(ByteBufferInterface io, PCAP parent, PCAP root) {
+
+            this.parent = parent;
+            this.root = root;
+            this.io = io;
+            read();
+        }
+
+        public Packet(long tSSec, long tSUsec, long inclLen, long origLen, byte[] rawBody, String network) {
+
+            this.tsSec = tSSec;
+            this.tsUsec = tSUsec;
+            this.inclLen = inclLen;
+            this.origLen = origLen;
+            this.raw_body = rawBody;
+        }
+
+        private void read() {
+            this.tsSec = this.io.readU4le();
+            this.tsUsec = this.io.readU4le();
+            this.inclLen = this.io.readU4le();
+            this.origLen = this.io.readU4le();
+            {
+                Linktype on = root().hdr().network();
+                if (on != null) {
+                    switch (root().hdr().network()) {
+                        case PPI: {
+                            this.raw_body = this.io.readBytes(
+                                    (inclLen() < root().hdr().snaplen() ? inclLen() : root().hdr().snaplen()));
+                            break;
+                        }
+                        case ETHERNET: {
+                            this.raw_body = this.io.readBytes(
+                                    (inclLen() < root().hdr().snaplen() ? inclLen() : root().hdr().snaplen()));
+                            break;
+                        }
```

Comment:
Intellij noted these are duplicate branches so they should be combined. In addition it suggested using Math.min instead of the tertiary operator.
```suggestion
                       case PPI, ETHERNET: {
                            this.raw_body = this.io.readBytes(
                                    (Math.min(inclLen(), root().hdr().snaplen())));
                            break;
                        }
```